django/core/checks/security/base.py

```diff
@@ -235,6 +239,65 @@ def check_referrer_policy(app_configs, **kwargs):
     return []
 
 
+@register(Tags.security, deploy=True)
+def check_cache_and_media_root(app_configs, **kwargs):
+    if not settings.MEDIA_ROOT:
+        return []
+    _media_root = os.path.abspath(settings.MEDIA_ROOT)
+    for alias in settings.CACHES.keys():
+        if isinstance(caches[alias], FileBasedCache):
+            location_check = (
+                _media_root,
+                caches[alias]._dir,
+            )
+            if os.path.commonpath(location_check) == _media_root:
+                return [Warning(
+                    "Your cache configuration might expose or corrupt your file system.",
+                    hint=f"Your cache '{alias}' is inside your 'MEDIA_ROOT'",
+                    id='security.W024',
+                )]
+    return []
+
+
+@register(Tags.security, deploy=True)
+def check_cache_and_static_root(app_configs, **kwargs):
+    if not settings.STATIC_ROOT:
+        return []
+    _static_root = os.path.abspath(settings.STATIC_ROOT)
+    for alias in settings.CACHES.keys():
+        if isinstance(caches[alias], FileBasedCache):
+            location_check = (
+                _static_root,
+                caches[alias]._dir,
+            )
+            if os.path.commonpath(location_check) == _static_root:
+                return [Warning(
+                    "Your cache configuration might expose or corrupt your file system.",
+                    hint=f"Your cache '{alias}' is inside your 'STATIC_ROOT'",
+                    id='security.W024',
+                )]
+    return []
+
+
+@register(Tags.security, deploy=True)
+def check_cache_and_staticfiles_dirs(app_configs, **kwargs):
+    if not settings.STATICFILES_DIRS:
+        return []
+    _abs_static_files_dirs = set()
+    for root in settings.STATICFILES_DIRS:
+        _abs_static_files_dirs.add(os.path.abspath(root))
+    for alias in settings.CACHES.keys():
+        if isinstance(caches[alias], FileBasedCache):
+            if any(os.path.commonpath((_root, caches[alias]._dir)) == _root
+                   for _root in _abs_static_files_dirs):
+                return [Warning(
+                    "Your cache configuration might expose or corrupt your file system.",
+                    hint=f"Your cache '{alias}' is inside your 'STATICFILES_DIRS'",
+                    id='security.W024',
+                )]
+    return []
```

Comment:
I think we should move this checks to `django/core/checks/caches.py`. Also, I would try to create a single check instead of three functions that are really similar.

Reply:
Just got this far down and see you've proposed the same.